Sources/DataPipeline/CIODataPipeline.swift

```diff
@@ -0,0 +1,173 @@
+import CioInternalCommon
+import Foundation
+import Segment
+
+public class CIODataPipeline {
+    // Private static instance of Analytics, created only once.
+    private static var analytics: Analytics = .init(configuration: Configuration.defaultConfiguration)
+
+    // Private initializer to enforce singleton usage, prevents external instantiation.
+    private init() {}
+
+    // Static method to access the singleton Analytics instance.
+    public static func shared() -> Analytics {
+        analytics
+    }
+
+    // Method to reinitialize Analytics with a new configuration.
+    public static func initialize(configuration: Configuration) {
+        analytics = Analytics(configuration: configuration)
+    }
+}
+
+// MARK: - System Modifiers
+
+public extension CustomerIO {
+    /// Enable/Disable analytics capture
+    var enabled: Bool {
+        CIODataPipeline.shared().enabled
+    }
+
+    /// Returns the anonymousId currently in use.
+    var anonymousId: String {
+        CIODataPipeline.shared().anonymousId
+    }
+
+    /// Returns the userId that was specified in the last identify call.
+    var userId: String? {
+        CIODataPipeline.shared().userId
+    }
+
+    /// Returns the current operating mode this instance was given.
+    var operatingMode: OperatingMode {
+        CIODataPipeline.shared().operatingMode
+    }
+
+    /// Adjusts the flush interval post configuration.
+    var flushInterval: TimeInterval {
+        CIODataPipeline.shared().flushInterval
+    }
+
+    /// Adjusts the flush-at count post configuration.
+    var flushAt: Int {
+        CIODataPipeline.shared().flushAt
+    }
+
+    /// Returns a list of currently active flush policies.
+    var flushPolicies: [FlushPolicy] {
+        CIODataPipeline.shared().flushPolicies
+    }
+
+    /// Returns the traits that were specified in the last identify call.
+    func traits<T: Codable>() -> T? {
+        CIODataPipeline.shared().traits()
+    }
+
+    /// Returns the traits that were specified in the last identify call, as a dictionary.
+    func traits() -> [String: Any]? {
+        CIODataPipeline.shared().traits()
+    }
+
+    /// Tells this instance of Analytics to flush any queued events up to Segment.com.  This command will also
+    /// be sent to each plugin present in the system.  A completion handler can be optionally given and will be
+    /// called when flush has completed.
+    func flush(completion: (() -> Void)? = nil) {
+        CIODataPipeline.shared().flush(completion: completion)
+    }
+
+    /// Resets this instance of Analytics to a clean slate.  Traits, UserID's, anonymousId, etc are all cleared or reset.  This
+    /// command will also be sent to each plugin present in the system.
+    func reset() {
+        CIODataPipeline.shared().reset()
+    }
+
+    /// Retrieve the version of this library in use.
+    /// - Returns: A string representing the version in "BREAKING.FEATURE.FIX" format.
+    func version() -> String {
```

Comment:
Do we want to expose the Segment SDK version? 

It feels weird to me for a customer to call `CustomerIO.version` and `CIODataPipeline.version` and get 2 different returned results? Also feels weird to expose the version of a dependency that our SDK has?

Reply:
I just feature matched the segments APIs, we can drop/keep or make any decisions in the `ready for production` tickets, otherwise the discussion might scope creep the PRs. What do you think?

Reply:
I am OK with deciding later. 

Idea: We make a decision in project slack channel? This seems like a short and quick decision to me. Could be answered async together.